services/src/pro/handlers/permissions.rs

```diff
@@ -0,0 +1,121 @@
+use actix_web::{web, FromRequest, HttpResponse};
+use serde::Deserialize;
+use utoipa::ToSchema;
+
+use crate::error::Result;
+
+use crate::pro::contexts::ProContext;
+use crate::pro::permissions::Permission;
+use crate::pro::permissions::{PermissionDb, ResourceId, RoleId};
+
+pub(crate) fn init_permissions_routes<C>(cfg: &mut web::ServiceConfig)
+where
+    C: ProContext,
+    C::Session: FromRequest,
+{
+    cfg.service(
+        web::scope("/permissions").service(
+            web::resource("")
+                .route(web::put().to(add_permission_handler::<C>))
+                .route(web::delete().to(remove_permission_handler::<C>)),
+        ),
+    );
+}
+
+/// Enum used for dispatching the permissions to the correct Dbs
+#[derive(Debug, PartialEq, Eq, Deserialize, Clone, Hash, ToSchema)]
+pub enum ResourceType {
+    // Dataset,
```

Comment:
so we need to add them? or are this suggestions?

Reply:
the type was redundant. The `ResourceId` is now used instead to differentiate the resource type. 

ad27b1d4bc4e9b1bc7ffb93e79cb6fd486679202